ratik/src/intTest/java/org/opencadc/ratik/InventoryValidatorTest.java

```diff
@@ -0,0 +1,855 @@
+        UUID artifactID = UUID.randomUUID();
+        URI artifactURI = URI.create("cadc:INTTEST/one.ext");
+
+        Artifact artifact = new Artifact(artifactID, artifactURI, TestUtil.getRandomMD5(),
+                                              new Date(), 1024L);
+        this.localEnvironment.artifactDAO.put(artifact);
+        this.remoteEnvironment.artifactDAO.put(artifact);
+
+        artifact.contentType = "image/fits";
+        this.remoteEnvironment.artifactDAO.put(artifact);
+
+        try {
+            System.setProperty("user.home", TMP_DIR);
+            InventoryValidator testSubject = new InventoryValidator(this.localEnvironment.daoConfig, TestUtil.LUSKAN_URI,
+                                                                    new IncludeArtifacts(),null,
+                                                                    trackSiteLocations);
+            testSubject.run();
+        } finally {
+            System.setProperty("user.home", USER_HOME);
+        }
+
+        Artifact localArtifact = this.localEnvironment.artifactDAO.get(artifact.getID());
+        Assert.assertNotNull("local artifact not found", localArtifact);
+        Assert.assertSame("local artifact metachecksum does not match",
+                          artifact.getMetaChecksum(), localArtifact.getMetaChecksum());
+    }
+
+    @Test
+    public void artifactChecksumMismatch_Explanation2_LocalIsStorage() throws Exception {
+        artifactChecksumMismatch_Explanation2(false);
+    }
+
+    @Test
+    public void artifactChecksumMismatch_Explanation2_LocalIsGlobal() throws Exception {
+        artifactChecksumMismatch_Explanation2(true);
+    }
+
+    public void artifactChecksumMismatch_Explanation2(boolean trackSiteLocations) throws Exception {
+        UUID artifactID = UUID.randomUUID();
+        URI artifactURI = URI.create("cadc:INTTEST/one.ext");
+
+        Artifact artifact = new Artifact(artifactID, artifactURI, TestUtil.getRandomMD5(),
+                                         new Date(), 1024L);
+        this.localEnvironment.artifactDAO.put(artifact);
+        this.remoteEnvironment.artifactDAO.put(artifact);
+
+        artifact.contentType = "image/fits";
+        this.localEnvironment.artifactDAO.put(artifact);
+
+        try {
+            System.setProperty("user.home", TMP_DIR);
+            InventoryValidator testSubject = new InventoryValidator(this.localEnvironment.daoConfig, TestUtil.LUSKAN_URI,
+                                                                    new IncludeArtifacts(),null,
+                                                                    trackSiteLocations);
+            testSubject.run();
+        } finally {
+            System.setProperty("user.home", USER_HOME);
+        }
+
+        Artifact localArtifact = this.localEnvironment.artifactDAO.get(artifact.getID());
+        Assert.assertNotNull("local artifact not found", localArtifact);
+        Assert.assertSame("local artifact metachecksum does not match",
```

Comment:
I prefer use of assertEquals... the check itself is sufficient, but subtle. This probably needs a comment like "local artifact is right and remote is wrong so discrepancy not be fixed"

Reply:
Updated to use assertEquals and updated the assert message.